Wangscape/utils.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <type_traits>
+#include <stdexcept>
+
+namespace
+{
+
+template<typename I>
+I _pow(I base, I exp)
```

Comment:
I think the decision to convert to/from wider datatypes before using `pow` should be up to the user code. Also, there are types with more digits than `long long` (`unsigned long long`, multiprecision types).